### PR TITLE
Added ctest LABELS

### PR DESCRIPTION
**Pull request by rafaelsamenezes**
Another minor step towards getting the infraestructure ready for #1637

1. Added directory and regression label to all regression tests.
2. Updated the documentation on how to invoke it.

### CODE REVIEW

**Review by lucasccordeiro — APPROVED** [review]
Excellent.

Thanks for submitting this PR, @rafaelsamenezes.

File: regression/README.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
 Here is an example to run a given regression suite: 
 
-```
-python3 testing_tool.py --regression floats --tool /Users/lucascordeiro/esbmc/build/src/esbmc/esbmc --mode CORE
-```
```
Comment (fbrausse) [inline regression/README.md:5]: Do we need to remove this command line example? I think it is useful to know how to run the Python script manually. `ctest` is just another way to invoke the script in an automated fashion.

Comment (rafaelsamenezes) [reply regression/README.md:5]: The test discovery of the testing tool was removed. So it is not working anymore.

Comment (fbrausse) [reply regression/README.md:5]: I see, thanks.

Comment (rafaelsamenezes) [reply regression/README.md:5]: It should be simple to add it back though. Let's just finish this test revamping first though.

**Pull request merged**